libtransact/src/scheduler/serial/core.rs

```diff
@@ -202,12 +202,10 @@ impl SchedulerCore {
                     self.try_schedule_next()?;
                 }
                 Ok(CoreMessage::ExecutionResult(task_notification)) => {
-                    let batch = match self.current_batch.take() {
-                        Some(batch) => batch,
-                        None => {
-                            panic!("received execution result but no current batch is executing")
-                        }
-                    };
+                    let batch = self
+                        .current_batch
+                        .take()
+                        .expect("received execution result but no current batch is executing");
```

Comment:
Given that this function returns a result, which in turn exits the thread with an error message on error, perhaps this should be propogated as an error, so as not to shutdown the whole application.

Reply:
This (and all other `expect`s other than lock acquisition) returns an error now